sklearn/compose/_column_transformer.py

```diff
@@ -417,6 +423,27 @@ def _validate_output(self, result):
                     "The output of the '{0}' transformer should be 2D (scipy "
                     "matrix, array, or pandas DataFrame).".format(name))
 
+    def _record_output_indices(self, Xs):
+        """
+        Record which transformer produced which column.
+        """
+        idx = 0
+        self.output_indices_ = {}
+
+        for transformer_idx, (name, _, _, _) in enumerate(
+            self._iter(fitted=True, replace_strings=True)
+        ):
+            n_columns = Xs[transformer_idx].shape[1]
+            self.output_indices_[name] = slice(idx, idx + n_columns)
+            idx += n_columns
+
+        # empty slices for transformers that generate no output (except
+        # for 'remainder' when 'drop'); these are safe for indexing
```

Comment:
About "except for remainer when drop": is this true? From the tests below, it seems that remainder is always part of the dict (as expected).